timemachine/fe/free_energy.py

```diff
@@ -550,16 +550,47 @@ def _combine(self, ligand_values: NDArray, host_values: Optional[NDArray] = None
 def get_water_sampler_params(initial_state: InitialState) -> NDArray:
     """Given an initial state, return a copy of the parameters that define the nonbonded parameters of water with respect to the
     entire system.
+
+    Water Interactions (with other Waters, Ligand, Protein) need to be consistent with the MD potentials.
+
+    Given the full N x N nonbonded interaction matrix:
+
+       P       W      L
+    ----------------------
+    |   P0 |   W0 |   L1 |
+    | P0   | P0   | P1   |
+    |------.------|-------
+    | P0   | W0   | W1   |
+    |   W0 |   W0 |   L1 |
+    ---------------------|
+    | P1   | W1   | L0   |
+    |   L1 |   L1 |   L0 |
+    ----------------------
+
+    Let W0, P0, L0 be the parameters from NonbondedAllPairs, W1, P1, L1 from the NonbondedInteractionGroup
+
+    We expect the parameters used for the water sampler to meet the following requirements:
+
+    1) Waters Parameters: assert that W0 == W1, we use NonbondedAllPairs but could use either
+    2) Protein Parameters: read protein parameters (P0) from NonbondedAllPairs
+    3) Ligand Parameters: read ligand parameters (L1) from NonbondedInteractionGroup
+
+    Important to note that P1 can change in Protein-Ligand charge fitting, which is okay because it is only expected
+    to impact Protein-Ligand interactions and not Protein-Water. If protein side chains are to be enhanced, the parameter
+    changes are expected to be in P0 to ensure correct water sampling.
     """
     nb_ixn_pot = get_bound_potential_by_type(initial_state.potentials, NonbondedInteractionGroup)
     water_params = np.array(nb_ixn_pot.params)
```

Comment:
nit: naming seems confusing since these are not all water parameters; should this be `nb_ixn_group_params` or similar?